liquibase-core/src/main/java/liquibase/lockservice/PostgresAdvisoryLockService.java

```diff
@@ -0,0 +1,173 @@
+package liquibase.lockservice;
+
+import liquibase.configuration.GlobalConfiguration;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.database.Database;
+import liquibase.database.core.PostgresDatabase;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LockException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.servicelocator.PrioritizedService;
+import liquibase.statement.core.RawSqlStatement;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.logging.Logger;
+
+public class PostgresAdvisoryLockService implements LockService {
+
+    private static final Logger LOGGER = Logger.getLogger(PostgresAdvisoryLockService.class.getName());
+
+    private static final int LOCK_NAMESPACE_ID = "liquibase".hashCode();
+    private static final int LOCK_ID = "lockservice".hashCode();
+
+    static final RawSqlStatement TRY_ACQUIRE_LOCK_SQL =
+        new RawSqlStatement(String.format("select pg_try_advisory_lock(%d,%d)", LOCK_NAMESPACE_ID, LOCK_ID));
+
+    static final RawSqlStatement RELEASE_LOCK_SQL =
+        new RawSqlStatement(String.format("select pg_advisory_unlock(%d,%d)", LOCK_NAMESPACE_ID, LOCK_ID));
```

Comment:
@RichardBradley I have updated the PR to use session advisory lock. 

I will run some more tests to confirm how session locks are released in case of an half-closed connection and document the behavior. 

Notice also that I have introduced a configuration flag `USE_DB_LOCK` that by default keep this lock strategy disabled so that if you want to use advisory locks instead of the standard lock you should explicitly enabled it.

Reply:
>  I have introduced a configuration flag USE_DB_LOCK that by default keep this lock strategy disabled

I think this should be on by default, as it is strictly better than the current locking system, if it is tested and working. It might be more likely to get merged if it is off by default, I suppose.